src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,30 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executor = Executors.newCachedThreadPool();
```

Comment:
Using Executors.newCachedThreadPool() can potentially create a large number of threads if many events are being notified simultaneously. Consider using Executors.newFixedThreadPool() with a specific number of threads or another type of ExecutorService that better suits your needs.